src/cfnlint/config.py

```diff
@@ -764,12 +764,11 @@ def _glob_filenames(self, filenames: Sequence[str]) -> list[str]:
 
         for filename in filenames:
             add_filenames = glob.glob(filename, recursive=True)
-            # only way to know of the glob failed is to test it
-            # then add the filename as requested
-            if not add_filenames:
-                all_filenames.append(filename)
-            else:
+
+            if isinstance(add_filenames, list):
                 all_filenames.extend(add_filenames)
+            else:
+                all_filenames.append(filename)
```

Comment:
not sure if this is still needed. I left in in here just in case there's an edge case I'm not aware of.

`glob.glob` should always return a list,